R/run_stress_test.R

```diff
@@ -366,3 +346,52 @@ iteration_sequence <- function(args_list) {
     return(seq_along(args_list[[iter_var]]))
   }
 }
+
+st_read <- function(input_path_project_specific, asset_type) {
+  path <- pacta_results_path(input_path_project_specific, asset_type)
+
+  out <- list(
+    pacta_results = read_pacta_results(path)
+  )
+
+  return(out)
+}
+
+st_process <- function(data, asset_type) {
+  pacta_results <- data$pacta_results %>%
+    process_pacta_results(
+      start_year = get_start_year(data),
+      end_year = end_year_lookup,
+      time_horizon = time_horizon_lookup,
+      scenario_geography_filter = scenario_geography_filter_lookup,
+      scenarios_filter = scenarios_filter(),
+      equity_market_filter = equity_market_filter_lookup,
+      sectors = sectors_lookup,
+      technologies = technologies_lookup,
+      allocation_method = allocation_method_lookup,
+      asset_type = asset_type
+    )
+
+  out <- list(
+    pacta_results = pacta_results
+  )
+
+  return(out)
+}
+
+pacta_results_path <- function(dir, asset_type) {
+  asset_type <- stringr::str_to_title(asset_type)
+  file <- glue::glue("{asset_type}_results_{calculation_level_lookup}.rda")
+  path <- file.path(dir, file)
+  return(path)
+}
```

Comment:
This serves two goals:
1. Abstract low level details like the exact name of the file.
2. Fit the creation of the path in 80 characters.